src/main/java/hudson/plugins/git/util/GitUtils.java

```diff
@@ -225,12 +242,12 @@ public static EnvVars getPollEnvironment(AbstractProject p, FilePath ws, Launche
                     }
                 }
             } else {
-                env = new EnvVars(System.getenv());
+                env = p.getEnvironment(workspaceToNode(ws), TaskListener.NULL);
             }
 
             p.getScm().buildEnvVars(b,env);
         } else {
-            env = new EnvVars(System.getenv());
+            env = p.getEnvironment(workspaceToNode(ws), listener);
```

Comment:
🐜 probably worth a try/catch and give the System.getEnv if it fails, since the workspace may be remote.